gson/src/main/java/com/google/gson/JsonPrimitive.java

```diff
@@ -292,4 +298,16 @@ private static boolean isIntegral(JsonPrimitive primitive) {
     }
     return false;
   }
+
+  private static boolean isBigInteger(JsonPrimitive primitive) {
+    return primitive.value instanceof BigDecimal;
```

Comment:
I think this should be `BigInteger`? Which in turn makes me think that there should be a test that would have caught this.

Speaking of tests, in general if you are proposing a fix for an issue, you should include a test that fails without the fix and passes with it.

Reply:
@eamonnmcmanus 
Yes that should be a BigInteger, as for the testing part, thank you for the tip, I will make sure to provide tests with any future commits.